### ISSUE TITLE
Multi-Viewport Issue when moving UI window 

### ISSUE THREAD

**Issue by AppledoreM** [ocornut/imgui#3455]
Version: 1.79
Branch: docking

Back-ends: imgui_impl_vulkan.cpp + imgui_impl_glfw.cpp 
Compiler: MSCV
Operating System: Windows 10

Hi, I am trying to incorporate ImGui into my own engine. I'm using the multi-viewport feature. However, every time I move an ImGui UI window out of the main window of my engine, the following error shows up:

>validation layer: Validation Error: [ VUID-vkCmdDrawIndexed-renderPass-02684 ] Object 0: handle = 0x81cab8000000009d, type = VK_OBJECT_TYPE_RENDER_PASS; Object 1: handle = 0xaff08a000000006f, type = VK_OBJECT_TYPE_RENDER_PASS; | MessageID = 0x8cb637c2 | vkCmdDrawIndexed(): RenderPasses incompatible between active render pass w/ VkRenderPass 0x81cab8000000009d[] and pipeline state object w/ VkRenderPass 0xaff08a000000006f[] Attachment 4294967295 is not compatible with 1: The first is unused while the second is not.. The Vulkan spec states: The current render pass must be compatible with the renderPass member of the VkGraphicsPipelineCreateInfo structure specified when creating the VkPipeline bound to VK_PIPELINE_BIND_POINT_GRAPHICS. (https://www.khronos.org/registry/vulkan/specs/1.1-extensions/html/vkspec.html#VUID-vkCmdDrawIndexed-renderPass-02684)

As long as the UI window stays in the main window, this error will not show up. I saw a similar error in this post [#3253](https://github.com/ocornut/imgui/issues/3253). I pretty new to vulkan, so I'm not sure what I should do here. Should I expect this to be fixed and just ignore this error for now?

**Comment by ocornut** [comment]
Hello,
If it looks the same as #3253 may add to that issue? A thoughtful fix would be welcome, I am also not much of a Vulkan users and don't have much bandwidth or experience to debug the use of this ever evolving SDK.

**Comment by AppledoreM** [comment]
It is not exactly the same though. That one is related to docking. But, I think it is reasonable to add to that issue. Should I just close this issue then?

**Comment by FunMiles** [comment]
@AppledoreM I think it is the same issue. You can try my pull request https://github.com/ocornut/imgui/pull/3459 and see if that fixes it.

**Comment by AppledoreM** [comment]
> @AppledoreM I think it is the same issue. You can try my pull request #3459 and see if that fixes it.

Yes!! I have just tested your pull request. It definitely fixes the bug. Thank you! 👍

**Comment by FunMiles** [comment]
I'm glad my PR worked for you. I guess it did work because you, like me, were creating your own full Vulkan state and passing it to the ImGui Vulkan implementation before doing anything else. 
However, my initial PR did break the demo. I've pushed a few more changes to allow the example to work. It keeps my own code, similar to yours working. So, once the PR is merged, I recommend you try the merged code. Or if you want, try it now and let me know.

**Comment by ocornut** [comment]
Merged the code in #3459 (with some changes, see there for details).
Thanks @FunMiles !
